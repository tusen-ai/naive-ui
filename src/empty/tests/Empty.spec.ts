import { mount } from '@vue/test-utils'
import { NEmpty } from '../index'

describe('n-empty', () => {
  it('should work with import on demand', () => {
    mount(NEmpty)
  })

  it('should work with slots', () => {
    const wrapper = mount(NEmpty, {
      slots: {
        default: () => 'test-description',
        icon: () => 'test-icon',
        extra: () => 'test-extra'
      }
    })

    expect(wrapper.find('.n-empty__description').exists()).toBe(true)
    expect(wrapper.find('.n-empty__description').text()).toBe(
      'test-description'
    )
    expect(wrapper.find('.n-empty__icon').exists()).toBe(true)
    expect(wrapper.find('.n-empty__icon').text()).toBe('test-icon')
    expect(wrapper.find('.n-empty__extra').exists()).toBe(true)
    expect(wrapper.find('.n-empty__extra').text()).toBe('test-extra')
    wrapper.unmount()
  })
  it('should work with `description` prop', () => {
    const wrapper = mount(NEmpty, {
      props: {
        description: 'test-description'
      }
    })

    expect(wrapper.find('.n-empty__description').exists()).toBe(true)
    expect(wrapper.find('.n-empty__description').text()).toContain(
      'test-description'
    )
    wrapper.unmount()
  })

  it('should work with `show-description` prop', async () => {
    const wrapper = mount(NEmpty)
    expect(wrapper.find('.n-empty__description').exists()).toBe(true)

    await wrapper.setProps({ showDescription: false })
    expect(wrapper.find('.n-empty__description').exists()).toBe(false)
    wrapper.unmount()
  })

  it('should work with `show-icon` prop', async () => {
    const wrapper = mount(NEmpty)
    expect(wrapper.find('.n-empty__icon').exists()).toBe(true)

    await wrapper.setProps({ showIcon: false })
    expect(wrapper.find('.n-empty__icon').exists()).toBe(false)
    wrapper.unmount()
  })

  it('should work with `size` prop', async () => {
    const wrapper = mount(NEmpty)

    expect(wrapper.find('.n-empty').attributes('style')).toMatchSnapshot()

    await wrapper.setProps({ size: 'small' })
    expect(wrapper.find('.n-empty').attributes('style')).toMatchSnapshot()

    await wrapper.setProps({ size: 'large' })
    expect(wrapper.find('.n-empty').attributes('style')).toMatchSnapshot()

    await wrapper.setProps({ size: 'huge' })
    expect(wrapper.find('.n-empty').attributes('style')).toMatchSnapshot()
    wrapper.unmount()
  })
})
