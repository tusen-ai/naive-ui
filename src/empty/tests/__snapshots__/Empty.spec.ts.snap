// Vitest Snapshot v1

exports[`n-empty > should work with \`size\` prop 1`] = `"--n-icon-size: 40px; --n-font-size: 14px; --n-bezier: cubic-bezier(.4, 0, .2, 1); --n-text-color: rgba(194, 194, 194, 1); --n-icon-color: rgba(194, 194, 194, 1); --n-extra-text-color: rgb(51, 54, 57);"`;

exports[`n-empty > should work with \`size\` prop 2`] = `"--n-icon-size: 34px; --n-font-size: 14px; --n-bezier: cubic-bezier(.4, 0, .2, 1); --n-text-color: rgba(194, 194, 194, 1); --n-icon-color: rgba(194, 194, 194, 1); --n-extra-text-color: rgb(51, 54, 57);"`;

exports[`n-empty > should work with \`size\` prop 3`] = `"--n-icon-size: 46px; --n-font-size: 15px; --n-bezier: cubic-bezier(.4, 0, .2, 1); --n-text-color: rgba(194, 194, 194, 1); --n-icon-color: rgba(194, 194, 194, 1); --n-extra-text-color: rgb(51, 54, 57);"`;

exports[`n-empty > should work with \`size\` prop 4`] = `"--n-icon-size: 52px; --n-font-size: 16px; --n-bezier: cubic-bezier(.4, 0, .2, 1); --n-text-color: rgba(194, 194, 194, 1); --n-icon-color: rgba(194, 194, 194, 1); --n-extra-text-color: rgb(51, 54, 57);"`;
